source/simulator/core.cc

```diff
@@ -1976,6 +1976,26 @@ namespace aspect
           break;
         }
 
+        case NonlinearSolver::Binary_input:
+        {
+          // Identical to IMPES except does not solve Stokes equation
+          if (parameters.free_surface_enabled)
+            free_surface->execute ();
+       
+          std::string fileName = parameters.input_directory + "/" + "solution-" + Utilities::int_to_string(timestep_number, 5) + ".mesh";
+          triangulation.load(fileName.c_str());
+          parallel::distributed::SolutionTransfer<dim, LinearAlgebra::BlockVector> sol_trans(dof_handler);
+          sol_trans.deserialize (solution);
+          //triangulation.clear();
```

Comment:
In terms of creating a snapshot or checkpoint files, this is done through the Boost archive classes. In other words,  the restart.resume.z file contains the state of the simulator.h, stokes_residual.h, point_values.h, visualization.h, and etc, since each of these classes implements the function `serialize (Archive &ar, const unsigned int)`. Is my understanding correct here? 

Is it possible to use boosts archive functionality to store only the time, time step, time step number, and etc, to a separate file at each time step?